File: src/_score/score_module.php
<?php

namespace Materia;

class Score_Modules_Sequencer extends Score_Module
{
	const ATTEMPT_PENALTY = 'attempt_penalty';

	public function check_answer($log)
	{
		if (isset($this->questions[$log->item_id]))
		{
			$q = $this->questions[$log->item_id];
			foreach ($q->answers as $answer)
			{
				if ($log->text == $answer['text']) return $answer['value'];
			}
		}

		if (isset($this->questions[$log->item_id]))
		{
			$q = $this->questions[$log->item_id];
			return $log->text == $q->answers[0]['text'] ? 100 : 0;
		}

		return 0;
	}

	protected function handle_log_widget_interaction($log)
	{
		if ($log->text == $this::ATTEMPT_PENALTY)
		{
			array_push($this->global_modifiers, $log->value);
		}
	}

}

File: src/_score/score_module.py
from scoring.module import ScoreModule


class Sequencer(ScoreModule):

    ATTEMPT_PENALTY = 'attempt_penalty'

    def check_answer(self, log):
        q = self.get_question_by_item_id(log.item_id)

        if q is not None:
            for answer in q["answers"]:
                if str(log.text) == str(answer["text"]):
                    return int(answer["value"])

        if q is not None:
            return 100 if log.text == q["answers"][0]["text"] else 0

        return 0

    def handle_log_widget_interaction(self, log):
        if log.text == self.ATTEMPT_PENALTY:
            self.global_modifiers.append(int(log.value))
